packages/components/tests/unit/chat/ChatSearch.spec.js

```diff
@@ -0,0 +1,143 @@
+
+    const resizeHandle = wrapper.find('[data-cy="resize-handle"]');
+    const resizeBy = 100;
+
+    resizeHandle.trigger('mousedown');
+    resizeHandle.trigger('mousemove', { clientX: 0 });
+    await wrapper.vm.$nextTick();
+    const initialWidth = Number.parseInt(lhsPanel.element.style.width.replace('px', ''), 10);
+
+    resizeHandle.trigger('mousemove', { clientX: resizeBy });
+    await wrapper.vm.$nextTick();
+    const newWidth = Number.parseInt(lhsPanel.element.style.width.replace('px', ''), 10);
+
+    expect(newWidth).toBe(initialWidth + resizeBy);
+  });
+
+  describe('when asked a question', () => {
+    let rpcFunction = (messages) => {
+      return (method, _, callback) => messages[method].shift()(callback);
+    };
+
+    it('makes expected RPC calls', async () => {
+      const searchResponse = {
+        results: [
+          {
+            appmap: 'example.appmap.json',
+            events: [],
+            score: 1.0,
+          },
+        ],
+      };
+      const messagesCalled = {
+        explain: [(cb) => cb(null, null, 'the-request-id')],
+        'explain.status': [
+          (cb) => cb(null, null, { step: 'build-vector-terms' }),
+          (cb) =>
+            cb(null, null, {
+              step: 'explain',
+              searchResponse,
+            }),
+          (cb) =>
+            cb(null, null, {
+              step: 'complete',
+              searchResponse,
+              explaination: 'Contact IT',
+            }),
+        ],
+        'appmap.metadata': [(cb) => cb(null, null, {})],
+        'appmap.data': [(cb) => cb(null, null, '{}')],
+      };
+      const wrapper = mount(VChatSearch, {
+        propsData: {
+          appmapRpcFn: rpcFunction(messagesCalled),
+        },
+      });
+
+      await wrapper.vm.sendMessage('How do I reset my password?');
+      await wrapper.vm.$nextTick();
+
+      Object.values(messagesCalled).forEach((calls) => {
+        // Callbacks are consumed on use, so we expect the array to be empty
+        expect(calls).toBeArrayOfSize(0);
+      });
+    });
+
+    describe('error handling', () => {
+      async function simulateError(err, error) {
+        const messagesCalled = {
+          explain: [(cb) => cb(err, error)],
+        };
+
+        const wrapper = mount(VChatSearch, {
+          propsData: {
+            appmapRpcFn: rpcFunction(messagesCalled),
+          },
+        });
+
+        await wrapper.vm.sendMessage('How do I reset my password?');
+        await wrapper.vm.$nextTick();
+
+        return wrapper;
+      }
+
+      describe('401 code', () => {
+        it('activates the login instructions', async () => {
+          const wrapper = await simulateError(undefined, { code: 401, message: 'Unauthorized' });
+          expect(wrapper.find('.status-unauthorized').exists()).toBe(true);
+          expect(wrapper.find('.messages .message').exists()).toBe(false);
+          expect(wrapper.find('.messages [data-error="true"]').exists()).toBe(false);
```

Comment:
Try to use `data-*` attributes as much as possible. We use the `data-cy` attribute as specific selector for test cases. From the [Cypress docs](https://docs.cypress.io/guides/references/best-practices#Selecting-Elements):

> Targeting the element above by `tag`, `class` or `id` is very volatile and highly subject to change. You may swap out the element, you may refactor CSS and update ID's, or you may add or remove classes that affect the style of the element.
>
> Instead, adding the `data-cy` attribute to the element gives us a targeted selector that's only used for testing.
>
> The `data-cy` attribute will not change from CSS style or JS behavioral changes, meaning it's not coupled to the behavior or styling of an element.
>
> Additionally, it makes it clear to everyone that this element is used directly by test code.

Reply:
Ok. In this case it seems like .messages was unlikely to change.